http/gateway.js

```diff
@@ -62,8 +57,9 @@ var HttpGateway = Extendable.extend({
 
                     if (response.statusCode >= 400) {
                         if (response.statusCode === 401) {
-                            dispatcher.emit('401-response-received');
+                            gateway.handle401(resolve, reject, method, path, data, headers);
```

Comment:
Good find. Not sure why, but it didn't seem to be causing problems on my testing. Perhaps the failure callbacks didn't do anything significant on the page being used to test.